src/main/scala/io/iohk/ethereum/vm/WorldStateProxy.scala

```diff
@@ -49,12 +49,26 @@ trait WorldStateProxy[WS <: WorldStateProxy[WS, S], S <: Storage[S]] { self: WS
   def transfer(from: Address, to: Address, value: UInt256): WS = {
     if(from == to) this
     else {
-      val debited = getGuaranteedAccount(from).increaseBalance(-value)
-      val credited = getAccount(to).getOrElse(getEmptyAccount).increaseBalance(value)
-      saveAccount(from, debited).saveAccount(to, credited)
+      guaranteedTransfer(from, to, value)
     }
   }
 
+  private def guaranteedTransfer(from: Address, to: Address, value: UInt256): WS = {
+    val debited = getGuaranteedAccount(from).increaseBalance(-value)
+    val credited = getAccount(to).getOrElse(getEmptyAccount).increaseBalance(value)
+    saveAccount(from, debited).saveAccount(to, credited)
+  }
+
+  /**
+    * In case of transfer to self, during selfdestruction the ether is actually destroyed
+    * see https://github.com/ethereum/wiki/wiki/Subtleties/d5d3583e1b0a53c7c49db2fa670fdd88aa7cabaf#other-operations
+    * and https://github.com/ethereum/go-ethereum/blob/ff9a8682323648266d5c73f4f4bce545d91edccb/core/state/statedb.go#L322
+    */
+  def removeEther(from: Address, value: UInt256): WS = {
```

Comment:
Maybe this should be `removeAllEther` with `address` parameter only? No other usages for it, are there?